src/content/insertable-streams/webgpu/js/multi-video.js

```diff
@@ -0,0 +1,245 @@
+        verticesBuffer.unmap();
+        this.verticesBuffer_ = verticesBuffer;
+
+        this.context_.configure({
+            device,
+            format: swapChainFormat
+        })
+
+        this.renderPipeline_ = device.createRenderPipeline({
+            vertex: {
+                module: device.createShaderModule({
+                    code: wgslShaders.vertex,
+                }),
+                entryPoint: 'main',
+                buffers: [
+                    {
+                        arrayStride: 20,
+                        attributes: [
+                            {
+                                // position
+                                shaderLocation: 0,
+                                offset: 0,
+                                format: 'float32x3',
+                            },
+                            {
+                                // uv
+                                shaderLocation: 1,
+                                offset: 12,
+                                format: 'float32x2',
+                            },
+                        ],
+                    },
+                ],
+            },
+            fragment: {
+                module: device.createShaderModule({
+                    code: wgslShaders.fragment,
+                }),
+                entryPoint: 'main',
+                targets: [
+                    {
+                        format: swapChainFormat,
+                    },
+                ],
+            },
+            primitive: {
+                topology: 'triangle-list',
+            },
+        });
+
+        this.videoTexture_ = device.createTexture({
+            size: {
+                width: 480 * 2,
+                height: 270 * 2,
+                depthOrArrayLayers: 1,
+            },
+            arrayLayerCount: 1,
+            mipLevelCount: 1,
+            sampleCount: 1,
+            dimension: '2d',
+            format: 'rgba8unorm',
+            usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.TEXTURE_BINDING |
+                GPUTextureUsage.RENDER_ATTACHMENT,
+        });
+
+    }
+
+    async transform(frame, frame2) {
+        const device = this.device_;
+        if (device == null) {
+            console.log("Device is null");
+            if (frame) frame.close();
+            if (frame2) frame2.close();
+            return;
+        }
+        // const canvas = this.canvas_;
+        const sampler = device.createSampler({
+            addressModeU: 'repeat',
+            addressModeV: 'repeat',
+            addressModeW: 'repeat',
+            magFilter: 'linear',
+            minFilter: 'linear',
+        });
+        const videoTexture = this.videoTexture_;
+        let videoFrame, videoFrame2;
+        if (frame) {
+            videoFrame = await createImageBitmap(frame, { resizeWidth: 480, resizeHeight: 270 });
+            device.queue.copyExternalImageToTexture(
```

Comment:
(No action required) Eventually we'll want to use the video textures directly. At that point, I assume we'll need to make quite a few changes to the shaders so that we can render from those textures.